crates/pathfinder/src/state/block_hash.rs

```diff
@@ -799,6 +878,57 @@ mod tests {
         assert_eq!(expected_final_hash, calculated_final_hash);
     }
 
+    /// Source:
+    /// https://github.com/starkware-libs/sequencer/blob/main/crates/starknet_api/src/block_hash/block_hash_calculator_test.rs#L74-121
+    #[rstest::rstest]
+    #[test]
+    fn test_final_transaction_hash_variants(
+        #[values(StarknetVersion::V_0_13_2, StarknetVersion::V_0_13_4)]
```

Comment:
Interesting! Didn't know about this.

Not sure if we may want this in two separate tests? cc/ @kkovaacs

Reply:
It's just copying what Starkware has...

Reply:
Under the hood `rstest` generates separate test cases so I think this is perfectly fine the way it is:

```
test state::block_hash::tests::test_final_transaction_hash_variants::starknet_version_1_StarknetVersion__V_0_13_2 ... ok
test state::block_hash::tests::test_final_transaction_hash_variants::starknet_version_2_StarknetVersion__V_0_13_4 ... ok
```

Reply:
I have a question here: what's the difference causing Pathfinder to need the `#[test]` attribute (on line 884) while https://github.com/starkware-libs/sequencer/blob/20845272bcb2c7c7bf9caa0347859febbecb2da1/crates/starknet_api/src/block_hash/block_hash_calculator_test.rs#L75 gets by without it? Or is the starkware-lib test actually not running? :-)

Reply:
I think `#[rstest]` implies `#[test]`: https://crates.io/crates/rstest

That means we should probably just remove `#[test]` from all our tests using rstest.